src/main/java/teammates/ui/pagedata/FeedbackSubmissionEditPageData.java

```diff
@@ -260,12 +291,112 @@ private FeedbackSubmissionEditQuestion createQuestion(FeedbackQuestionAttributes
+                                questionAttributes, "", getResponseVisibilityMap(questionAttributes, true), giverName,
+                                recipientName, false, sessionTimeZone);
+                    } else {
+                        giverName = getPreviewInstructor().name;
+                        responseExplanationComment = buildFeedbackResponseCommentAddForm(
+                                questionAttributes, "", getResponseVisibilityMap(questionAttributes, false), giverName,
+                                recipientName, true, sessionTimeZone);
+                    }
+                }
+                responses.add(new FeedbackSubmissionEditResponse(responseIndx, false, recipientOptionsForQuestion,
+                        submissionFormHtml, "", responseExplanationComment));
+                i++;
+            } else {
+                responses.add(new FeedbackSubmissionEditResponse(responseIndx, false, recipientOptionsForQuestion,
+                        submissionFormHtml, ""));
+            }
             responseIndx++;
         }
 
         return responses;
     }
+
+    private List<String> getRecipientList(List<String> responseSubmittedRecipient, Map<String, String> sortedRecipientList) {
+        List<String> recipientList = new ArrayList<String>();
+        for (String recipient : sortedRecipientList.values()) {
+            if (!responseSubmittedRecipient.contains(recipient)) {
+                recipientList.add(recipient);
+            }
+        }
+        return recipientList;
+    }
+
+    private List<FeedbackResponseCommentRow> getResponseCommentsForQuestion(FeedbackQuestionAttributes question,
+            List<FeedbackResponseCommentAttributes> frcList, String giverName,
+            String recipientName, Map<FeedbackParticipantType, Boolean> responseVisibilityMap,
+            Map<String, String> commentGiverEmailNameTable) {
+        List<FeedbackResponseCommentRow> frcCommentRowList = new ArrayList<FeedbackResponseCommentRow>();
+        double sessionTimeZone = bundle.feedbackSession.getTimeZone();
+        List<FeedbackResponseCommentAttributes> filteredFrcs =
+                filterFeedbackResponseCommentAttributes(bundle.roster, frcList);
+        for (FeedbackResponseCommentAttributes frcAttributes : filteredFrcs) {
+            FeedbackResponseCommentRow frc = new FeedbackResponseCommentRow(frcAttributes, frcAttributes.giverEmail,
+                    giverName, recipientName, getResponseCommentVisibilityString(frcAttributes, question),
+                    getResponseCommentGiverNameVisibilityString(frcAttributes, question),
+                    responseVisibilityMap, commentGiverEmailNameTable, sessionTimeZone);
+            setEditDeleteCommentOptionForUser(frcAttributes, frc);
+            frcCommentRowList.add(frc);
+        }
+        return frcCommentRowList;
+    }
+
+    private List<FeedbackResponseCommentAttributes> filterFeedbackResponseCommentAttributes(CourseRoster roster,
```

Comment:
State what this method is trying to filter by, preferably in the method name, but can use a javadoc instead if it is too long.

Reply:
I missed this.Will fixed this in a while